src/libs/elektra/kdb.c

```diff
@@ -1874,6 +1889,23 @@ int kdbGet (KDB * handle, KeySet * ks, Key * parentKey)
 		clear_bit (parentKey->flags, KEY_LOCK_NAME | KEY_LOCK_VALUE);
 	*/
 
+	// Step 13: run gopts (if enabled)
+	keyCopy (parentKey, initialParent, KEY_CP_NAME);
```

Comment:
As stated in the other comment, you should do:

```suggestion
	keyCopy (parentKey, initialParent, KEY_CP_NAME);
	keySetNamespace (parentKey, KEY_NS_CASCADING);
```

and then before step 15 and you may need

```c
keySetNamespace (parentKey, keyGetNamespace (initialParent));
```